### DISCUSSION TITLE
Using Specific Time with Take Until

### DISCUSSION THREAD

**Discussion by kaelabruin** [bonsai-rx/bonsai#2024] [Q&A] [answered]
Hi,
[autostart_dates_01.n_cams=4.cams=1-4.crf=28.fps=50.v08.zip](https://github.com/user-attachments/files/17106119/autostart_dates_01.n_cams.4.cams.1-4.crf.28.fps.50.v08.zip)

I want to edit my workflow to automatically stop every day at 11:59 pm. Is there a way to do this in the workflow. I know I can use TakeUntil with a timer, but could I use TakeUntil with a specific time? Preferably I'd want to do this within Bonsai, but I am also open to other solutions.

Thanks!

**Comment by bruno-f-cruz** [comment, marked as answer]
Use a `Timer` that ticks periodically (adjust `Period` to get the resolution you are looking for). Once it goes over the time of the day, use the TakeUntil as you mentioned.
eg.:

![image](https://github.com/user-attachments/assets/077f0065-10b2-49a2-b0e9-72114a1ae95c)

**Reply by kaelabruin** [reply]
This worked perfectly, thank you!
